### PR TITLE
Microphones Component implementation

### PR DESCRIPTION
**Pull request by fdelcampo**

### CODE REVIEW

File: love/src/components/Mics/Table/Row.jsx [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+  };
+
+  /* Method to change the isSelected state, by the mics component*/
+  selectMe = () => {
+    const { isSelected } = this.state;
+    if (!isSelected) {
+      this.setState({ isSelected: true, alarm: false });
+    } else {
+      this.setState({ isSelected: !isSelected });
+    }
```
Comment (sebastian-aranda) [inline love/src/components/Mics/Table/Row.jsx:328]: You should use the following setState form here instead:
```javascript
this.setState((prevState) => {
  if (!prevState.isSelected) {
    return { isSelected: true, alarm: false };
  }
  else {
    return { isSelected: false };
  }
});
```

File: love/src/components/Mics/Table/Row.jsx [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+    this.props.setInfoPlot(infoPlot);
+  };
+
+  /**
+   * Chnage the notification's state
+   */
+  turnNotifications = () => {
+    const { notifications } = this.state;
+    if (notifications) this.setState({ notifications: false, alarm: false });
+    else this.setState({ notifications: true });
```
Comment (sebastian-aranda) [inline love/src/components/Mics/Table/Row.jsx:351]: You should use the following setState form here instead:
```javascript
this.setState((prevState) => {
  if (prevState.notifications) {
    return { notifications: false, alarm: false };
  }
  else {
    return { notifications: true };
  }
});
```

File: love/src/components/Mics/Table/Table.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    selectMic: PropTypes.func,
+    /**
+     * Function to add a new record on the mic's record state
+     */
+    recordPush: PropTypes.func,
+    /**
+     * Function to set the infoPlot state of the mic component to render
+     */
+    setInfoPlot: PropTypes.func,
+    initialPlaying: PropTypes.bool,
```
Comment (sebastian-aranda) [inline love/src/components/Mics/Table/Table.jsx:32]: Please add a description for the `initialPlaying` propType 🙏️.

**Review by sebastian-aranda — CHANGES_REQUESTED** [review]
Great job @fdelcampo, thanks! I've left some comments, please check those out 🙏️.

**Review by sebastian-aranda — CHANGES_REQUESTED** [review]
Great job @fdelcampo, thanks! I've left some comments, please check those out 🙏️.

File: love/src/components/Mics/Details/DrawerMic.jsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    isRecording: PropTypes.bool,
+    /**
+     * Function to start or stop record
+     */
+    record: PropTypes.func,
+    /**
+     * Array of records made previously
+     */
+    records: PropTypes.array,
+    maxHeight: PropTypes.number,
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/components/Mics/Details/DrawerMic.jsx:58]: Removed maxHeight prop. Not necessary the definition

File: love/src/components/Mics/Details/HeatMap/HeatMap.jsx [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    if (!this.props.infoPlot) {
+      return <></>;
+    }
+    const { data3D } = this.props.infoPlot;
+    return (
+      <>
+        {/* To resize, its really important preserve the structure since it is enunciated the Heat Map tag
+      in the div with the container ref until this part. Then, to this work, NOT include anothers div on here,
+      only which will enclose the Vega Lite Plot tag.*/}
+        {/* <div className={styles.divVegaLite}> */}
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/components/Mics/Details/HeatMap/HeatMap.jsx:224]: removed

**Review by sebastian-aranda — APPROVED** [review]
Great job @fdelcampo, I love how the component ended! Thanks for reviewing the feedback, go ahead with the merge 👍️.

**Comment by sebastian-aranda** [comment]
You will need a rebase.

**Pull request merged**